File: src/0905-Sort-Array-by-Parity/sort_array_by_parity.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var sortArrayByParity = function(nums) {
  let even_index = 0

  for (let i = 0; i < nums.length; i++) {
    if (nums[i] % 2 === 0) {
      [nums[i], nums[even_index]] = [nums[even_index], nums[i]]
      even_index++
    }
  }
    return nums
}

nums = [3,1,2,4]

console.log(sortArrayByParity(nums))

File: src/0905-Sort-Array-by-Parity/sort_array_by_parity.py
from typing import List

class Solution:
    def sortArrayByParity(self, nums: List[int]) -> List[int]:
        even_index: int = 0

        for i in range(len(nums)):
            if nums[i] % 2 == 0:
                nums[i], nums[even_index] = nums[even_index], nums[i]
                even_index += 1
        return nums

if __name__ == "__main__":
    nums: List[int] = [3,1,2,4]
    
    solution = Solution()
    
    print(solution.sortArrayByParity(nums))